back-end/helpers/user-helpers.js

```diff
@@ -18,6 +18,42 @@ async function getUserGitHubInfo(userID) {
   return userGitHubData;
 }
 
+async function getUserLanguage(userID) {
+  const query = 'SELECT language FROM github_info WHERE user_id = $1';
+  const userProfile = await queryWithParameters(
+      query,
+      [userID]
+  );
+  if(userProfile === undefined){
+    return false;
+  }else {
+    return userProfile.language;
+  }
+}
+
+async function getRecommendations(userID) {
+  const languageToMatch = await getUserLanguage(userID);
+  if(languageToMatch) {
+    // user_id clause so they are not recommended to connect with themself.
+    // ordered randomly so that same recommendations don't happen every time.
+    const query = 'SELECT id, email, phone, twitter FROM users INNER JOIN' +
```

Comment:
Even though this query will be changed, I think it's best for now to only return their email and id. I believe it was a customer requirement to only show additional details of a user after they are connected.

Reply:
Updated

Reply:
Should we be showing any private information to users who are not yet connected? I imagined we might show the matching user's github name, since that is publicly-discoverable, vs. email, which is not.

Reply:
This might also help users decided whether they want to connect to the user or not.